pandas/core/arrays/_mixins.py

```diff
@@ -304,6 +304,68 @@ def __repr__(self) -> str:
     # ------------------------------------------------------------------------
     # __array_function__ methods
 
+    def __array_function__(self, func, types, args, kwargs):
+        for x in types:
+            if not issubclass(x, (np.ndarray, NDArrayBackedExtensionArray)):
+                return NotImplemented
+
+        if not args:
+            # TODO: if this fails, are we bound for a RecursionError?
+            for key, value in kwargs.items():
+                if value is self:
+                    # See if we can treat self as the first arg
+                    import inspect
+
+                    sig = inspect.signature(func)
+                    params = sig.parameters
+                    first_argname = next(iter(params))
+                    if first_argname == key:
+                        args = (value,)
+                        del kwargs[key]
+                        break
+                    else:
+                        kwargs[key] = np.asarray(self)
+                        break
+
+        if args and args[0] is self:
+
+            if func in [np.delete, np.repeat, np.atleast_2d]:
+                res_data = func(self._ndarray, *args[1:], **kwargs)
+                return self._from_backing_data(res_data)
```

Comment:
How do we know that there's 1 output argument from `func` here? Do we have a `func.nout` or something like that to check?

Reply:
we don't, thats why we're implementing only a specific handful of functions here